### PR TITLE
Redesign repository overview

### PR DESCRIPTION
**Pull request by eheimbuch**
## Proposed changes

Change repository overview layout to use single rows instead cards. Also remove quick links and add clone action to repository entry. The default repository link now leads to the sources view.

### Your checklist for this pull request

**Contributor**:
- [x] PR is well described and the description can be used as a commit message on squash
- [x] Related issues linked to PR if existing and labels set
- [ ] New code is covered with unit tests
- [x] New ui components are tested inside the storybook (module ui-components only) 
- [x] [Changelog entry file](https://github.com/scm-manager/changelog#changelog-entry-files) created in `gradle/changelog` or CHANGELOG.md is updated for plugins
- [ ] Feature has been tested with different permissions
- [ ] Documentation updated (only necessary for new features or changed behaviour)

**Reviewer**:
- [ ] The clean code principles are respected ([CleanCode](https://clean-code-developer.com/virtues/))
- [ ] All new code/logic is implemented on the right spot / "Should this be done here?"
- [ ] UI changes fits into the layout
- [ ] The UI views / components are responsive (mobile views)
- [ ] Correct translations are available

### Checklist for branch merge request (not required for forks)

- [ ] Branch is green/blue on [Jenkins](https://oss.cloudogu.com/jenkins/)
- [ ] Quality Gate passed on [SonarQube](https://sonarcloud.io/organizations/scm-manager/projects)

### CODE REVIEW

File: scm-ui/ui-components/src/layout/GroupEntry.tsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  padding: 0.5rem;
+  align-items: center;
+  pointer-events: all;
+`;
+
+const OverlayLink = styled(Link)`
+  width: 100%;
+  position: absolute;
+  height: calc(90px - 1.5rem);
+  pointer-events: all;
```
Comment (sdorra) [inline scm-ui/ui-components/src/layout/GroupEntry.tsx:43]: The overlay looks much nicer with a border radius `border-radius: 4px`

File: scm-ui/ui-components/src/repos/RepositoryAvatar.tsx [hunk 2]
```diff
@@ -56,7 +56,7 @@ const renderExtensionPoint = (repository: Repository) => {
 };
 
 const RepositoryAvatar: FC<Props> = ({ repository }) => {
-  return <Avatar className="image is-64x64">{renderExtensionPoint(repository)}</Avatar>;
+  return <Avatar className="image is-48x48">{renderExtensionPoint(repository)}</Avatar>;
```
Comment (sdorra) [inline scm-ui/ui-components/src/repos/RepositoryAvatar.tsx:59]: I don't think that it is a good idea to change the default size of the repository avatar. We should make the size configurable and let the default untouched.

See https://github.com/scm-manager/scm-manager/pull/1738/commits/f65ca6ad0825baf62b478078aeee6d4e05c1ae33

File: scm-ui/ui-components/src/repos/RepositoryEntry.tsx [hunk 3]
```diff
@@ -21,207 +21,118 @@
...
-          {t("repository.healthCheckFailure")}
-        </RepositoryFlag>
-      );
-    }
+const RepositoryEntry: FC<Props> = ({ repository, baseDate }) => {
+  const [t] = useTranslation("repos");
+  const [openCloneModal, setOpenCloneModal] = useState(false);
 
+  const createContentRight = () => {
     return (
```
Comment (sdorra) [inline scm-ui/ui-components/src/repos/RepositoryEntry.tsx:83]: Unnecessary `return`

File: scm-ui/ui-components/src/repos/RepositoryFlag.tsx [hunk 4]
```diff
@@ -35,7 +35,7 @@ type Props = {
 };
 
 const RepositoryFlag: FC<Props> = ({ children, title, size = "small", ...props }) => (
-  <Tooltip location="bottom" message={title}>
+  <Tooltip location="right" message={title}>
```
Comment (sdorra) [inline scm-ui/ui-components/src/repos/RepositoryFlag.tsx:38]: This change breaks at least the labels of the repository mirror log page.

Comment (sdorra) [reply scm-ui/ui-components/src/repos/RepositoryFlag.tsx:38]: I think we should make it configurable and keep the default.

File: scm-ui/ui-components/src/repos/RepositoryFlags.tsx [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+import RepositoryFlag from "./RepositoryFlag";
+import { ExtensionPoint } from "@scm-manager/ui-extensions";
+import { Repository } from "@scm-manager/ui-types";
+import { useTranslation } from "react-i18next";
+import styled from "styled-components";
+import HealthCheckFailureDetail from "./HealthCheckFailureDetail";
+
+type Props = {
+  repository: Repository;
+  className?: string;
```
Comment (sdorra) [inline scm-ui/ui-components/src/repos/RepositoryFlags.tsx:35]: I think we need a prop for the orientation of the flag tooltips. That we can use right orientation in the repository overview and keep bottom for the rest.

File: scm-ui/ui-webapp/src/repos/containers/RepositoryRoot.tsx [hunk 6]
```diff
@@ -120,7 +119,7 @@ const RepositoryRoot = () => {
   if (redirectUrlFactory) {
     redirectedUrl = url + redirectUrlFactory(props);
   } else {
-    redirectedUrl = url + "/info";
+    redirectedUrl = url + "/code/sources/";
```
Comment (sdorra) [inline scm-ui/ui-webapp/src/repos/containers/RepositoryRoot.tsx:122]: Linking directly to the source code makes sense most of the time. But it does not make sense for the redirect after the repository was created. I see two options for the prolem:

1. Redirect to the information after repository creation
2. Add checkout/clone information to the code page, if the repository is empty

**Review by sdorra — CHANGES_REQUESTED** [review]
There are some problems if the landing page plugin is installed:

1. Single Repository View: Repository flags are now below the fav star
2. Repository Overview: Fav star is not clickable

File: scm-ui/ui-webapp/src/repos/containers/RepositoryRoot.tsx (hunk 6 above)
Comment (eheimbuch) [reply scm-ui/ui-webapp/src/repos/containers/RepositoryRoot.tsx:122]: i prefer the first solution

File: scm-ui/ui-components/src/repos/RepositoryEntry.tsx [hunk 7]
```diff
@@ -21,207 +21,116 @@
...
+    <div className="is-flex">
+      <ExtensionPoint name="repository.card.beforeTitle" props={{ repository }} />
+      <Name>{repository.name}</Name> <RepositoryFlags repository={repository} className="is-hidden-mobile" />
+    </div>
+  );
+
+  return (
+    <>
+      <GroupEntry
+        avatar={<RepositoryAvatar repository={repository} />}
```
Comment (sdorra) [inline scm-ui/ui-components/src/repos/RepositoryEntry.tsx:126]:
```suggestion
        avatar={<RepositoryAvatar repository={repository} size={48} />}
```

File: scm-ui/ui-components/src/Tooltip.tsx [hunk 8]
```diff
@@ -31,9 +31,11 @@ type Props = {
   children: ReactNode;
 };
 
+export type TooltipLocation = "bottom" | "right" | "top" | "left";
```
Comment (sdorra) [inline scm-ui/ui-components/src/Tooltip.tsx:34]: Why is the type not used for the location prop?

File: scm-ui/ui-components/src/repos/RepositoryFlags.tsx [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+import { Repository } from "@scm-manager/ui-types";
+import { useTranslation } from "react-i18next";
+import styled from "styled-components";
+import HealthCheckFailureDetail from "./HealthCheckFailureDetail";
+import { TooltipLocation } from "../Tooltip";
+
+type Props = {
+  repository: Repository;
+  className?: string;
+  tooltipLocations?: TooltipLocation;
```
Comment (sdorra) [inline scm-ui/ui-components/src/repos/RepositoryFlags.tsx:37]: I think we should not use the plural, because it is always a single tooltip.

**Review by sdorra — CHANGES_REQUESTED** [review]

File: scm-ui/ui-components/src/Tooltip.tsx (hunk 8 above)
Comment (eheimbuch) [reply scm-ui/ui-components/src/Tooltip.tsx:34]: I was not sure if this would be a breaking change. I know it shouldn't be but it kinda seems like one.

Comment (sdorra) [reply scm-ui/ui-components/src/Tooltip.tsx:34]: Changing a type definition is not a breaking change. Type are not relevant for the runtime at all. They are only hints for the developer and this change makes the hint only better and does not change anything at runtime.

**Pull request merged**